platform/darwin/src/MGLPointCollection.mm

```diff
@@ -68,7 +68,7 @@ - (MGLCoordinateBounds)overlayBounds
 
 - (BOOL)intersectsOverlayBounds:(MGLCoordinateBounds)overlayBounds
 {
-    return MGLLatLngBoundsFromCoordinateBounds(_bounds).intersects(MGLLatLngBoundsFromCoordinateBounds(overlayBounds));
+    return MGLCoordinateBoundsIntersectsCoordinateBounds(_bounds, overlayBounds);
```

Comment:
> In general, I wonder if it'd ever be useful to add an `intersectsOverlay: <MGLOverlay>` API for all of these objects that conform to `MGLOverlay`?

This SDK isn’t a general-purpose geometry library, so I think we should only add geometric methods as we need them or if we need them for parity with MapKit. In the future, we may want to spin out these geometry classes as part of a separate library that would serve as a more natural home for such geometry functions.